### PR TITLE
Case List Explorer case property labels

### PR DESCRIPTION
**Pull request by pxwxnvermx**
## Product Description
This feature adds user-defined labels for each selected case property.These display labels would allow users to view the report data in a more readable format and provide a better user experience. 
image
image

## Technical Summary
[Ticket Link](https://dimagi-dev.atlassian.net/browse/INDIV-64)

## Feature Flag
https://staging.commcarehq.org/hq/flags/edit/case_list_explorer/

### Safety story
Tested this PR locally and on staging.

### QA Plan

[QA Ticket](https://dimagi-dev.atlassian.net/browse/QA-4694)

### Rollback instructions

This feature makes changes to the config for explorer columns. To rollback this feature additional code need to be added in order to keep the saved reports with labels functional.
Rollback Code #32411 

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/standard/cases/filters.py
Comment (mkangia) [inline corehq/apps/reports/standard/cases/filters.py:133]: :+1:

**Comment by orangejenny** [comment]
Not reviewing, just noting this is a nice usability improvement 🚀

**Review by mkangia — APPROVED** [review]
Looks great @pxwxnvermx.

Should we modify tests to now use the new syntax and may be add a new one with legacy format to make sure it is not forgotten?

Good to merge post QA.

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (mkangia) [inline corehq/apps/reports/tests/test_case_list_explorer_report.py:30]: oh wow, that needed quite the setup. How about using `corehq.apps.users.tests.test_auditors.MockRequest`?

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (mkangia) [inline corehq/apps/reports/tests/test_case_list_explorer_report.py:39]: nit: short names like this, do add an extra overhead later for anyone looking at the code. Just `query_dict` or `get_query_dict` would make it clear.

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (mkangia) [inline corehq/apps/reports/tests/test_case_list_explorer_report.py:42]: Is there a reason to use setList earlier but not here?

**Review by mkangia — COMMENTED** [review]
Looks great!

Going to wait for QA for approval though since the impact is high.

**Comment by mkangia** [comment]
@pxwxnvermx 
for the PR description,

Since there are no migrations, seems good to remove the check for migrations compatibility.

Anything concerning about rollback if needed?

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (pxwxnvermx) [reply corehq/apps/reports/tests/test_case_list_explorer_report.py:42]: CaseListExplorerColumns expects an encoded json string and setList didn't work as it directly provided a list.

https://github.com/dimagi/commcare-hq/blob/baf502f34802765640d82e7792f63f1c67f1ea72/corehq/apps/reports/standard/cases/filters.py#L124

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (pxwxnvermx) [reply corehq/apps/reports/tests/test_case_list_explorer_report.py:30]: I tried `MockRequest` it does not replicate whole request interface and lacks http `method`, so `request.GET` doesn't work with it.

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (mkangia) [reply corehq/apps/reports/tests/test_case_list_explorer_report.py:42]: Okay, thank you. Good to leave a comment in the code for this though, its hard to guess that.

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (mkangia) [reply corehq/apps/reports/tests/test_case_list_explorer_report.py:30]: How hard would it be to modify `MockRequest` to support what you need?
This would help others to use `MockRequest` if they need to mock requests and avoid re-inventing the wheel each time.

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (esoergel) [inline corehq/apps/reports/tests/test_case_list_explorer_report.py:33]: Looks like you're missing `cls.domain.delete()` here

File: corehq/apps/reports/tests/test_case_list_explorer_report.py
Comment (pxwxnvermx) [reply corehq/apps/reports/tests/test_case_list_explorer_report.py:30]: The underlying view is expecting a whole request object, hence the changes here don't seem feasible to add to the `MockRequest` class, `RequestFactory` generates a request object for testing so it can be changed accordingly for requirements of the test.

**Review by mkangia — APPROVED** [review]
Good to go after QA

**Comment by mkangia** [comment]
@pxwxnvermx reminder to delete branches once merged. Its recommended to avoid having stale branches around.

**Pull request merged**
